### PR TITLE
Fix issues with django and jinja2 exceptions with just-my-code turned on. Fixes #1181

### LINKED ISSUE #1181: Verify django and jinja2 exceptions context
See: https://github.com/Microsoft/ptvsd/pull/1179/files/90dc0b17a0bceadbc0dad958be04ad5f2fa81404#diff-a3b5595e743c389ebd7b61a7ab4c3c80R131

### PR DESCRIPTION
**Pull request by fabioz**

### CODE REVIEW

File: src/ptvsd/_vendored/pydevd/tests_python/test_debugger_json.py [hunk 1]
```diff
@@ -919,16 +923,16 @@ def test_stepping(case_setup):
...
-            pydevd_schema.StepInRequest(pydevd_schema.StepInArguments(hit.thread_id)))
-        stepout_response = json_facade.wait_for_response(stepout_request)
-        hit = writer.wait_for_breakpoint_hit(REASON_STEP_RETURN)
-
-        stack_trace_request = json_facade.write_request(
-            pydevd_schema.StackTraceRequest(pydevd_schema.StackTraceArguments(threadId=hit.thread_id)))
-        stack_trace_response = json_facade.wait_for_response(stack_trace_request)
-        stack_frame = next(iter(stack_trace_response.body.stackFrames))
-        assert stack_frame['name'] == 'Call'
+#         stepout_request = json_facade.write_request(
```
Comment (fabioz) [inline src/ptvsd/_vendored/pydevd/tests_python/test_debugger_json.py]: @karthiknadig 
Note: this part of the test was actually failing (it seems that the step back is still not implemented in `pydevd`?), so, just commented out for now.

Comment (karthiknadig) [reply src/ptvsd/_vendored/pydevd/tests_python/test_debugger_json.py]: I think it should be StepOut. Looks like i had a typo in my PR.

File: src/ptvsd/_vendored/pydevd/pydevd_plugins/django_debug.py [hunk 2]
```diff
@@ -346,8 +356,21 @@ def _is_django_variable_does_not_exist_exception_break_context(frame):
...
 
-def can_not_skip(plugin, main_debugger, frame):
-    return main_debugger.django_breakpoints and _is_django_render_call(frame)
+def can_skip(plugin, main_debugger, frame):
+    if main_debugger.django_breakpoints:
+        if _is_django_render_call(frame):
+            return False
+
+    if main_debugger.django_exception_break:
+        module_name = frame.f_globals.get('__qualname__', '')
```
Comment (int19h) [inline src/ptvsd/_vendored/pydevd/pydevd_plugins/django_debug.py]: I don't think modules can have `__qualname__`. There was a PEP for that, but it was withdrawn.

File: src/ptvsd/_vendored/pydevd/pydevd_plugins/django_debug.py [hunk 3]
```diff
@@ -223,6 +222,13 @@ def _get_source_django_18_or_lower(frame):
         return None
 
 
+def _convert_to_str(s):
+    if IS_PY2:
+        if isinstance(s, unicode):
+            s = s.encode('utf-8', 'replace')
```
Comment (int19h) [inline src/ptvsd/_vendored/pydevd/pydevd_plugins/django_debug.py]: Shouldn't need `'replace'`, since UTF-8 can always represent any Unicode string.

**Pull request merged**